addons/mail/static/src/components/attachment/attachment.js

```diff
@@ -103,18 +108,18 @@ class Attachment extends Component {
      * @param {MouseEvent} ev
      */
     _onClickDownload(ev) {
-        ev.stopPropagation();
+        markEventHandled(ev, 'Attachment.actionButton');
         this.env.services.navigate(`/web/content/ir.attachment/${this.attachment.id}/datas`, { download: true });
     }
 
     /**
-     * Open the attachment viewer when clicking on viewable attachment.
+     * Open the attachment viewer when clicking on viewable ²attachment.
      *
      * @private
      * @param {MouseEvent} ev
      */
     _onClickImage(ev) {
-        if (!this.attachment.isViewable) {
+        if (!this.attachment.isViewable || isEventHandled(ev, 'Attachment.actionButton')) {
```

Comment:
I'm not fond of || in conditional for preemptive return. It makes code harder to read.

I prefer this:

```js
if (!this.attachment.isViewable) {
    return;
}
if (isEventHandled(ev, 'Attachment.clickDownload')) {
    return;
}
if (isEventHandled(ev, 'Attachment.clickUnlink')) {
    return;
}
```